server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -46,17 +46,25 @@ public class TimeSeriesIdFieldMapper extends MetadataFieldMapper {
     public static final String NAME = "_tsid";
     public static final String CONTENT_TYPE = "_tsid";
     public static final TimeSeriesIdFieldType FIELD_TYPE = new TimeSeriesIdFieldType();
-    public static final TimeSeriesIdFieldMapper INSTANCE = new TimeSeriesIdFieldMapper();
```

Comment:
The reason why this is a singleton constant is to reduce memory usage of `MapperService`.  Meta field mappers that can be initialized as a constant are a constant. A node can have many `MapperService` instances (one per allocated index).

I think can just define two singleton instances here and then decide to use which in `IndexMode#timeSeriesIdFieldMapper(...)` based on index versions and whether the feature flag is enabled?